pkg/didcomm/common/service/service.go

```diff
@@ -34,6 +41,35 @@ type DIDCommMsg struct {
 	ToVerKeys []string
 }
 
+// ThreadID returns msg ~thread.thid if there is no ~thread.thid returns msg @id
+// message is invalid if ~thread.thid exist and @id is absent
+func (m *DIDCommMsg) ThreadID() (string, error) {
+	msg := struct {
+		ID     string           `json:"@id"`
+		Thread decorator.Thread `json:"~thread"`
+	}{}
+
+	if err := json.Unmarshal(m.Payload, &msg); err != nil {
```

Comment:
Will each decorator need to do its own Unmarshal? @rolsonquadras @llorllale @soluchok